rockcraft/project.py

```diff
@@ -33,17 +33,36 @@ class ProjectValidationError(CraftError):
     """Error validatiing rockcraft.yaml."""
 
 
+class SupportInfo(pydantic.BaseModel):
+    """Schema for the 'support' property"""
+    end_of_life: Optional[str] = pydantic.Field(alias='end-of-life')
+    end_of_support: Optional[str] = pydantic.Field(alias='end-of-support')
+    info: Optional[str]
+    
+    
 class Project(pydantic.BaseModel):
     """Rockcraft project definition."""
 
+    # required
     name: str
-    version: str
     base: Literal["ubuntu:18.04", "ubuntu:20.04"]
-    build_base: Optional[str]
-    entrypoint: Optional[List[str]]
+    description: str
+    license: str
+    parts: Dict[str, Any]
+    summary: str
+    version: str
+    # optional
+    build_base: Optional[str] = pydantic.Field(alias='build-base')
     cmd: Optional[List[str]]
+    contact: Optional[List[str]]
+    documentation: Optional[str]
```

Comment:
Just a note that in `metadata.yaml` for charms, this field is called `docs` - may be worth sticking with it for consistency?

Reply:
sounds good. changed